app/src/test/kotlin/com/wire/android/util/DateTimeUtilKtTest.kt

```diff
@@ -18,124 +18,201 @@
 
 package com.wire.android.util
 
+import kotlinx.datetime.toKotlinInstant
+import org.junit.jupiter.api.AfterAll
 import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
 
 class DateTimeUtilKtTest {
 
-    @Test
-    fun `given a invalid date, when performing a transformation, then return null`() {
-        val result = "NOT_VALID".deviceDateTimeFormat()
-        assertEquals(null, result)
+    @Nested
+    @DisplayName("DateAndTimeParser Should")
+    inner class DateAndTimeParsers {
+        @Test
+        fun `return null when an invalid date`() {
+            val result = "NOT_VALID".deviceDateTimeFormat()
+            assertEquals(null, result)
+        }
+
+        @Test
+        fun `return a medium format date, when a valid date`() {
+            val result = "2022-03-24T18:02:30.360Z".deviceDateTimeFormat()
+            assertEquals("March 24, 2022, 6:02 PM", result)
+        }
+
+        @Test
+        fun `return a medium format, when formatMediumDateTime is called`() {
+            val result = "2022-03-24T18:02:30.360Z".formatMediumDateTime()
+            assertEquals("Mar 24, 2022, 6:02:30 PM", result)
```

Comment:
These tests fail if the JVM running them doesn't have `US` as the default locale. For instance, if I set my computer's language to German:

```
org.opentest4j.AssertionFailedError: expected: <March 24, 2022, 6:02 PM> but was: <24. März 2022, 18:02>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1145)
	at com.wire.android.util.DateTimeUtilKtTest$DateAndTimeParsers.return a medium format date, when a valid date(DateTimeUtilKtTest.kt:47)
```

### Ideally

It would be nice to allow receiving the `Locale` as parameter in these functions:
Maybe leave `Locale.getDefault()` as default argument. And rewriting the to have a backing property:
```kotlin
@Stable
fun String.deviceDateTimeFormat(locale: Locale = Locale.getDefault()): String? = DateAndTimeParsers.deviceDateTimeFormat(this)
```

But at least we can write the tests like so:

```kotlin
        @Test
        fun `return a medium format date, when a valid date`() {
            val result = "2022-03-24T18:02:30.360Z".deviceDateTimeFormat(Locale.US)
            assertEquals("March 24, 2022, 6:02 PM", result)
        }
```

### Shortcut

Maybe we can just set the default Locale before the test.

```kotlin
        @Test
        fun `return a medium format date, when a valid date`() {
            Locale.setDefault(Locale.US)
            val result = "2022-03-24T18:02:30.360Z".deviceDateTimeFormat()
            assertEquals("March 24, 2022, 6:02 PM", result)
        }
```

Reply:
The  test you mention is not failing anymore 😅  it was solved, with the `@BeforeAll`, `@AfterAll` blocks to set the Locale, but funny enough, the old `java.text.DateFormat`, the implementation we are deprecating, is not respecting this, so even the shortcut does not works. 

The test failing is for retro compatibility 

```
return the same MediumOnlyDateTime format result, when calling date with new DateTimeFormatter format
```

Reply:
But I agree we should find a better way to be more reliable on the tests in all context, so I would love to have it passing the locale to the parser 👍

Reply:
Oooh indeed. I missed that in the tests. Didn't see the `@BeforeAll` and `@AfterAll` down there.